Spigot-Server-Patches/0615-Optimized-villager-pathfinding-and-added-option-to-s.patch

```diff
@@ -0,0 +1,293 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: lukas81298 <lukas81298@gommehd.net>
+Date: Sat, 12 Dec 2020 01:59:27 +0100
+Subject: [PATCH] Optimized villager pathfinding and added option to slow down villagers
+ 
+ The villager pathfinding searches for structures in villages a villager might navigate to (e.g. lecterns). By default, this scans all chunk sections in a horizontal range of 48 blocks. The
+ vanilla implementation uses a very complex chain of streams and contains lots of redundant computations and checks way more chunk sections than actually necessary. This patch contains
+ a rewritten version of this algorithm which should be about 10 times faster and reduces the pressure on the garbage collector. Using the default configuration, the behavior should not differ from 
+ vanilla, however lowering "villageSearchRadius" in the "paper.yml" should boost the performance significantly with just a little inpact on the gameplay.
+ Furthermore, this patch adds the option to modify the "cooldown" between consecutive pathfinding calls to the same villager by modifying "villagerBehaviorMinCooldown" and
+ "villagerBehaviorMaxCooldown". The cooldown will be a random number of ticks between these two values. The default values do not change the vanilla behavior.
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index 6ab0a51b421dc696aec292f2db955311c9cc31e3..15e5b70508caede9981d249b5fe634773fba53a8 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -462,4 +462,17 @@ public class PaperConfig {
+     private static void trackPluginScoreboards() {
+         trackPluginScoreboards = getBoolean("settings.track-plugin-scoreboards", false);
+     }
++
++    public static int villageSearchRadius = 48;
++    private static void villagerLookForPlaceChunkRange() {
++        villageSearchRadius = getInt("settings.village-search-radius", 48);
++    }
++
++    public static int villagerBehaviorMinCooldown = 0;
++    public static int villagerBehaviorMaxCooldown = 20;
++
++    private static void villagerBehaviorCooldown() {
++        villagerBehaviorMinCooldown = getInt("settings.villager-behavior-cooldown-min", 0);
++        villagerBehaviorMaxCooldown = getInt("settings.villager-behavior-cooldown-max", 20);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/BehaviorFindPosition.java b/src/main/java/net/minecraft/server/BehaviorFindPosition.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..129da93b63fa77c0e05c812cf7882f2072e2aedb
+--- /dev/null
++++ b/src/main/java/net/minecraft/server/BehaviorFindPosition.java
```

Comment:
Looks like you forgot to import this file using the mcdev import script.

https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#i-cant-find-the-nms-file-i-need